### ISSUE TITLE
Define API(s) that expose dataset descriptions

### ISSUE THREAD

**Issue by ddeboer** [CLARIAH/clariah-plus#67]
The API(s) should support searching and retrieving dataset descriptions.

- SPARQL?
- Elasticsearch?

Ineo should become a client of this API.

**Comment by menzowindhouwer** [comment]
If we go for an existing repository/registry, e.g. CKAN, Dataverse or Fedora, there will, in general, be an API already ...

**Comment by jblom** [comment]
@menzowindhouwer @ddeboer yes... very good point: it seems we first need to investigate which of these existing registries is already suitable for our needs. Should I create an issue for that? (can't seem to find one)

Amongst other things (such as good support), the system should at least have a good API so clients can easily search/retrieve datasets. Also it would be great if it's possible to extend the system with a way of publishing to different dataset models (e.g. based on schema.org or DCAT or the NDE dataset model for cultural heritage collections).

It would be nice if we can avoid building a registry ourselves, since we do not have that much time.

In the B&G/Media Suite [search API](https://github.com/beeldengeluid/labs-search-api) (still a private repo, hope to open it up soon) we also have a `collection_registry` endpoint that can harvest collections from CKAN (or any other registry if you implement a plugin). It should not be too hard to extend that endpoint with a way to generate different publishing models as well (and implement [content negotiation by profile](https://w3c.github.io/dx-connegp/connegp/), which @wmelder also implemented for our B&G open datasets endpoint).

**Comment by menzowindhouwer** [comment]
@jblom created #90 to discuss existing registries or using an faceted index

**Comment by wmelder** [comment]
We use a very old version of CKAN for internal use in the Media Suite, and use the [simple API](http://docs.ckan.org/en/2.8/api/) provided. But this API seems not adequate for Clariah purposes. New CKAN version including [DCAT extension](https://extensions.ckan.org/extension/dcat/) seem promising. It contains dataset endpoints and harvesting facilities. None of this has been installed or tested on B&G side yet. Nor do we participate in the CKAN development. Currently, we provide [open dataset descriptions](https://data.beeldengeluid.nl/id/datacatalog/0001) in schema.org on our beng-lod-server. It is used only for open datasets published by B&G.
